src/pudl/transform/eia860.py

```diff
@@ -941,3 +941,70 @@ def clean_boiler_emissions_control_equipment_assn_eia860(
     bece_df = bece_df.dropna(subset="emission_control_id_eia")
 
     return bece_df
+
+
+@asset
+def clean_boiler_cooler_assn_eia860(
+    raw_boiler_cooling_eia860: pd.DataFrame,
+) -> pd.DataFrame:
+    """Pull and transform the EIA 860 boiler to cooler ID table.
+
+    Args:
+        raw_boiler_booling_eia860: Raw EIA 860 boiler to cooler ID association table.
+
+    Returns:
+        pd.DataFrame: A cleaned and normalized version of the EIA boiler to cooler ID
+            table.
+    """
+    # Replace empty strings, whitespace, and '.' fields with real NA values
+    bc_assn = pudl.helpers.fix_eia_na(raw_boiler_cooling_eia860)
+    # Replace the report year col with a report date col for the harvesting process
+    bc_assn = pudl.helpers.convert_to_date(
+        df=bc_assn, year_col="report_year", date_col="report_date"
+    )
+    # Drop rows with no cooling ID and just in case, drop duplicate
+    bc_assn = bc_assn.dropna(subset="cooling_id_eia").drop_duplicates()
+
+    return bc_assn
+
+
+@asset
+def clean_boiler_stack_flue_assn_eia860(
+    raw_boiler_stack_flue_eia860: pd.DataFrame,
+) -> pd.DataFrame:
+    """Pull and transform the EIA 860 boiler to stack flue ID table.
+
+    Args:
+        raw_boiler_booling_eia860: Raw EIA 860 boiler to stack flue ID association
+            table.
+
+    Returns:
+        pd.DataFrame: A cleaned and normalized version of the EIA boiler to stack flue
+            ID table.
+    """
+    # Replace empty strings, whitespace, and '.' fields with real NA values
+    bsf_assn = pudl.helpers.fix_eia_na(raw_boiler_stack_flue_eia860)
+    # Replace the report year col with a report date col for the harvesting process
+    bsf_assn = pudl.helpers.convert_to_date(
+        df=bsf_assn, year_col="report_year", date_col="report_date"
+    )
+    # Drop duplicates
+    bsf_assn = bsf_assn.drop_duplicates()
+    # Create a primary key column for stack flue IDs.
+    # There are some stack & flue units that are combined and have just one ID. Others
+    # have unique IDs for each stack & flue unit that may have m:m relationships with
+    # one another. These IDs are part of the primary key for this table, and because we
+    # can't have NA values in the primary key we'll need to create a hybrid column that
+    # turns the seperate stack_id_eia and flue_id_eia values into one stack_flue_id_pudl
+    # value. Anytime there's a stack_flue_id_eia we'll use that.
+    bsf_assn = bsf_assn.assign(
+        stack_flue_id_pudl=lambda x: (
+            x.stack_id_eia.astype("string") + "_" + x.flue_id_eia.astype("string")
+        )
+        .fillna(x.stack_flue_id_eia)
```

Comment:
There aren't any overlaps in the two sets of columns, correct? Otherwise I would suggest prioritizing filling with the combined ID over the two separate ID columns by reordering the order of assignment.

Reply:
Yup! there is no overlap between the two.

Reply:
If there's never any overlap between these two ways of identifying the stacks and flues -- each record has either a stack ID and a flue ID *OR* a stack-and-flue ID, is there any reason why we wouldn't want to assign null values of both the stack ID and flue ID the stack-and-flue ID?  That would avoid creating a new composite `stack_flue_id_pudl` column and also ensure that both `stack_id_eia` and `flue_id_eia` are always non-null.  And it seems like it would preserve the general idea of the stack ID and flue ID without losing any information.

Reply:
This only works insofar as there are always values for `stack_id_eia` and `flue_id_eia`, but it looks like there are 15 rows with no `flue_id_eia`, only a `stack_id_eia`.

Reply:
Wow that's annoying. 15 records out of how many?  Do we think that's accurate or bad data? It would be much simpler / more similar to all of our other data if we just had these IDs in two nice clean columns. @e-belfer what do you think about this?

Reply:
Also, they don't report `stack_id_eia` and `flue_id_eia` anymore, so we could just take `stack_flue_id_eia` and split it into both `flue_id_eia` and `stack_id_eia`, but A) that means conserving old columns that aren't used anymore, and B) potentially recording false `flue_id_eia` values.